pages/[...slug].tsx

```diff
@@ -141,63 +138,70 @@ const DocSlugs = ({ source, frontMatter, pagination }: Props) => {
 export default DocSlugs;
 
 export const getStaticProps = async ({ params }) => {
-    // Absolute path of the docs file
-    const postFilePath = path.join(DOCS_PATH, `${path.join(...params.slug)}.mdx`);
-    // Raw Mdx File Data Buffer
-    const source = fs.readFileSync(postFilePath);
-
-    /**
-     * Content: Mdx Data
-     * data: FrontMatter Data
-     */
-    const { content, data } = matter(source);
-
-    const allDocs = getAllDocs();
-    const nav = getNavfromDocs(allDocs);
-    const [currentDocSlug] = params.slug as string[];
-    const currentDocs = allDocs.filter((doc) => doc.url.includes(`/${currentDocSlug}/`));
-    const { previousPost, nextPost } = getPagination(currentDocs, params.slug as string[]);
-    const pagination = { previousPost, nextPost };
-    const toc = [];
-    const mdxSource = await renderToString(content, {
-        components,
-        // Optionally pass remark/rehype plugins
-        mdxOptions: {
-            remarkPlugins: [
-                require('@/lib/remark-code-header'),
-                require('@fec/remark-a11y-emoji'),
-                withTableofContents(toc),
-                imagePlugin
-            ],
-            rehypePlugins: [mdxPrism]
-        },
-        scope: data
-    });
-    return {
-        props: {
-            toc,
-            pagination,
-            nav: { [currentDocSlug]: nav[currentDocSlug] },
-            source: mdxSource, // { compiledSource: mdxSource.compiledSource },
-            frontMatter: data,
+    try {
+        if (params.slug.join('/') !== params.slug.join('/').toLowerCase())
+            return {
+                redirect: {
+                    destination: `/${params.slug.join('/').toLowerCase()}`,
+                    permanent: true,
+                },
+            }
+        const url = !process.env.VERCEL_URL || process.env.VERCEL_URL === 'localhost' ? `http://localhost:${process.env.PORT}` : `https://${process.env.VERCEL_URL}`;
+        const { docs: allDocs, nav } = (await (await fetch(`${url}/docs/api/content?query=*`)).json())
+        const [currentDocSlug] = params.slug as string[];
+        const currentDocs = allDocs.filter((doc) => doc.url.includes(`/${currentDocSlug}/`));
+        const { previousPost, nextPost } = getPagination(currentDocs, params.slug as string[]);
+        const pagination = { previousPost, nextPost };
+        const toc = [];
+        const [currentNavDoc] = allDocs.filter((doc) => doc.url.includes(`/${join(...params.slug)}`));
+        const { content } = currentNavDoc
+        const data = {
+            ...currentNavDoc
         }
-    };
+        delete data.content;
+        const mdxSource = await renderToString(content, {
```

Comment:
do we think doing this at run time might cause performance issue after deploying docs. 

as all the pages will be fresh and will go through this flow to generate pages ?